kiliautoml/models/_hugging_face_text_classification_model.py

```diff
@@ -216,6 +228,81 @@ def _compute_asset_classification(model_framework, tokenizer, model, asset):
+        for met in metrics:
+            if met == "accuracy":
+                metric_res[met] = metric[met].compute(predictions=predictions, references=labels)[
+                    met
+                ]
+            elif met == "f1":
+                metric_res[met] = np.append(
+                    metric[met].compute(predictions=predictions, references=labels, average=None)[
+                        met
+                    ],
+                    metric[met].compute(
+                        predictions=predictions, references=labels, average="weighted"
+                    )[met],
+                )
+            else:
+                metric_res[met] = np.append(
+                    metric[met].compute(
+                        predictions=predictions,
+                        references=labels,
+                        average=None,
+                        zero_division=0,
+                    )[met],
+                    metric[met].compute(
+                        predictions=predictions,
+                        references=labels,
+                        average="weighted",
+                        zero_division=0,
+                    )[met],
+                )
+        return metric_res
+
+    def model_evaluation(self, trainer, job_categories):
+        train_metrics = trainer.evaluate(trainer.train_dataset)
+        val_metrics = trainer.evaluate()
+        model_evaluation: Dict[str, Any] = {}
+
+        if len(train_metrics["eval_precision"]) == len(job_categories) + 1:
+            for i, label in enumerate(job_categories):
+                train_label_metrics = {}
+                for metric in ["precision", "recall", "f1"]:
+                    train_label_metrics[metric] = train_metrics["eval_" + metric][i]  # type: ignore
+                model_evaluation["train_" + label] = train_label_metrics
+        if len(val_metrics["eval_precision"]) == len(job_categories) + 1:
+            for i, label in enumerate(job_categories):
+                val_label_metrics = {}
+                for metric in ["precision", "recall", "f1"]:
+                    val_label_metrics[metric] = val_metrics["eval_" + metric][i]  # type: ignore
+                model_evaluation["val_" + label] = val_label_metrics
+        model_evaluation["train__overall"] = {
+            "loss": train_metrics["eval_loss"],
+            "accuracy": train_metrics["eval_accuracy"],
+            "precision": train_metrics["eval_precision"][-1],  # type: ignore
+            "recall": train_metrics["eval_recall"][-1],  # type: ignore
+            "f1": train_metrics["eval_f1"][-1],  # type: ignore
+        }
+        model_evaluation["val__overall"] = {
+            "loss": val_metrics["eval_loss"],
+            "accuracy": val_metrics["eval_accuracy"],
+            "precision": val_metrics["eval_precision"][-1],  # type: ignore
+            "recall": val_metrics["eval_recall"][-1],  # type: ignore
+            "f1": val_metrics["eval_f1"][-1],  # type: ignore
```

Comment:
I don't understand why sometimes there is a [-1] and sometimes not

Reply:
-1 correspond to the average metric (add comment)

Reply:
add Dataclass